nncf/tensorflow/nncf_network.py

```diff
@@ -0,0 +1,37 @@
+
+import tensorflow as tf
+
+
+class NNCFNetwork(tf.keras.Model):
+    def __init__(self, target_model):
+        super().__init__(target_model)
+        self.nncf_wrapped_model = target_model
+        self.kd_original_model = None
+        self.kd_outputs_storage = None
+        self.kd_loss = tf.keras.losses.MeanSquaredError()
+
+    def get_nncf_wrapped_model_config(self):
+        return self.nncf_wrapped_model.get_config()
+
+    def get_config(self):
+        return self.nncf_wrapped_model.get_config()
+
+    def compile(self, *args, **kwargs):
+        super(NNCFNetwork, self).compile(*args, **kwargs)
+        self.nncf_wrapped_model.compile(*args, **kwargs)
+
+    def call(self, inputs, training=None, mask=None):
+        compressed_output = self.nncf_wrapped_model(inputs)
+        if self.kd_original_model is not None:
```

Comment:
You could manage the train/eval graph by training parameter. For example
```suggestion
        if self.kd_original_model is not None and training:
```
Now during evaluation tf build graph consisting only of target model